pkg/elasticsearch/search.go

```diff
@@ -0,0 +1,99 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"text/template"
+
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/flanksource/commons/logger"
+	"github.com/flanksource/flanksource-ui/apm-hub/api/logs"
+)
+
+type ElasticSearchBackend struct {
+	client   *elasticsearch.Client
+	template *template.Template
+	index    string
+}
+
+func NewElasticSearchBackend(client *elasticsearch.Client, index, tpl string) (*ElasticSearchBackend, error) {
+	logger.Infof("Parsing template: %s", tpl)
+
+	template, err := template.New("query").Parse(tpl)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ElasticSearchBackend{
+		client:   client,
+		index:    index,
+		template: template,
+	}, nil
+}
+
+func (t *ElasticSearchBackend) Search(q *logs.SearchParams) (logs.SearchResults, error) {
+	var result logs.SearchResults
+	var buf bytes.Buffer
+	if err := t.template.Execute(&buf, q.Labels); err != nil {
```

Comment:
@moshloop I'm using Labels from the search param as the view for now.
I don't think any of the fields that we currently have in `logs.SearchParams` is suitable for this use case.
Should I add another field?